WalletWasabi/Blockchain/Keys/KeyManager.cs

```diff
@@ -429,68 +383,29 @@ public void SetDoNotSkipSynchronization()
 			({ } k, { } i) => GetKeys(x => x.IsInternal == i && x.KeyState == k)
 		};
 
-	/// <param name="ignoreTail">If true it does only consider the gap between used keys and does not care about the nonused keys at the end.</param>
-	public int CountConsecutiveUnusedKeys(bool isInternal, bool ignoreTail)
+	public HdPubKeyPathView GetView(bool isInternal, ScriptPubKeyType scriptPubKeyType)
 	{
-		var keyIndexes = GetKeys(x => x.IsInternal == isInternal && x.KeyState != KeyState.Used).Select(x => x.Index).ToArray();
-
-		if (ignoreTail)
+		var keySource = GetHdPubKeyGenerator(isInternal, scriptPubKeyType);
+		lock (CriticalStateLock)
 		{
-			var lastUsedIndex = GetKeys(x => x.IsInternal == isInternal && x.KeyState == KeyState.Used).LastOrDefault()?.Index;
-			if (lastUsedIndex is null)
-			{
-				return 0;
-			}
-			else
-			{
-				keyIndexes = keyIndexes.Where(x => x < lastUsedIndex).ToArray();
-			}
-		}
-
-		var hs = keyIndexes.ToHashSet();
-		int largerConsecutiveSequence = 0;
-
-		for (int i = 0; i < keyIndexes.Length; ++i)
-		{
-			if (!hs.Contains(keyIndexes[i] - 1))
-			{
-				int j = keyIndexes[i];
-				while (hs.Contains(j))
-				{
-					j++;
-				}
-
-				var sequenceLength = j - keyIndexes[i];
-				if (largerConsecutiveSequence < sequenceLength)
-				{
-					largerConsecutiveSequence = sequenceLength;
-				}
-			}
+			return HdPubKeyCache.GetView(keySource.KeyPath);
```

Comment:
I don't see how it could happen. It is only called by a method that we have to delete and even if we don't delete the method it calls this method with parameters that make `keySource` never null. 

Let's wait. I don't want to add more things, it is already too big, hard to break in smaller parts and nobody really understands what the keymanager does so, I would prefer to rely on testing.